include/seqan3/alphabet/cigar/cigar.hpp

```diff
@@ -0,0 +1,409 @@
+    {
+        return seqan3::to_char(get<0>(*this));
+    }
+
+    //!\brief Return the length value. This reads the internal length value.
+    constexpr length_type to_length() const noexcept
+    {
+        return get<1>(*this);
+    }
+
+    //!\brief Return the cigar element as a string. This reads both, the internal seqan3::cigar_op letter and length value.
+    string_type to_string() const noexcept
+    {
+        std::string tmp{std::to_string(get<1>(*this))};
+        tmp.push_back(seqan3::to_char(get<0>(*this)));
+        return tmp;
+    }
+    //!\}
+};
+
+//!\brief Type deduction guide enables usage of cigar without specifying template arguments.
+//!\relates cigar
+template <typename length_type>
+cigar(cigar_op &&, length_type &&)
+    -> cigar<std::decay_t<length_type>>;
+
+//!\brief Type deduction guide enables usage of cigar without specifying template arguments.
+//!\relates cigar
+template <typename length_type>
+cigar(cigar_op const &, length_type &&)
+    -> cigar<std::decay_t<length_type>>;
+
+/*!\brief Specialized implementation of seqan3::alphabet_concept::operator<<()
+ *        for the seqan3::cigar alphabet. Delegates to seqan3::cigar::to_string().
+ * \relates cigar
+ * \tparam length_type Type of the length value; must model std::UnsignedIntegral.
+ * \param  os          The output stream you are printing to.
+ * \param  cig         The seqan3::cigar element that you wish to print.
+ * \returns A reference to the output stream.
+ */
+template<typename length_type>
+//!\cond
+    requires std::UnsignedIntegral<length_type>
+//!\endcond
+std::ostream & operator<<(std::ostream & os, cigar<length_type> const & cig)
+{
+    os << cig.to_string();
+    return os;
+}
+
+// ------------------------------------------------------------------
+// containers
+// ------------------------------------------------------------------
+
+//!\brief Alias for a std::vector over the default seqan3::cigar alphabet.
+//!\ingroup cigar
+using cigar_vector = std::vector<cigar<>>;
+
+/*!\brief Specialized implementation of the stream operator for a container of
+ *        seqan3::cigar elements. Delegates to seqan3::cigar::to_string().
+ * \relates cigar
+ * \tparam length_type Type of the length value; must model std::UnsignedIntegral.
+ * \tparam range_type  Must model std::ranges::ForwardRange and its
+ *                     reference type must be seqan3::cigar<length_type>.
+ * \param  os          The output stream you are printing to.
+ * \param  cigv        The container of cigar elements.
+ * \returns A reference to the output stream.
+ */
+template <typename length_type = uint32_t, std::ranges::ForwardRange range_type>
+//!\cond
+    requires std::UnsignedIntegral<length_type> &&
+             std::Same<reference_t<range_type>, cigar<length_type> &>
+//!\endcond
+std::ostream & operator<<(std::ostream & os, range_type const & cigv)
+{
+    std::for_each(cigv.begin(), cigv.end(), [&os] (auto cig) { os << cig; } );
+    return os;
+}
```

Comment:
we shouldn't have this here at this point, instead we should add a generic one for debug_stream